helpers/helperUtils.ts

```diff
@@ -95,3 +90,93 @@ export function getProviders() {
     { network: NETWORK_NAMES.LINEA_MAINNET, rpcUrl: process.env.LINEA_MAINNET_RPC_URL as string },
   ];
 }
+
+function getUnifiedContractAddress(contractName: string): string {
+  let contractProperty;
+  for (const property in CONTRACT_NAMES) {
+    if (CONTRACT_NAMES[property] === contractName) {
+      contractProperty = property;
+      break;
+    }
+  }
+  return UNIFIED_CONTRACT_ADDRESSES[contractProperty];
+}
+
+export async function getPoseidonN(nInputs: number): Promise<Contract | null> {
+  const abi = poseidonContract.generateABI(nInputs);
+  const contractAddress = getUnifiedContractAddress(`PoseidonUnit${nInputs}L`);
+
+  if (!(await isContract(contractAddress))) {
+    return null;
+  }
+  const poseidon = new ethers.Contract(contractAddress, abi);
+
+  return poseidon;
+}
+
+export async function getUnifiedContract(contractName: string): Promise<Contract | null> {
+  if (contractName.includes("PoseidonUnit")) {
+    const nInputs = parseInt(contractName.substring(12, 13));
+    return getPoseidonN(nInputs);
+  } else {
+    const contractAddress = getUnifiedContractAddress(contractName);
+    if (!(await isContract(contractAddress))) {
+      return null;
+    }
+    return ethers.getContractAt(contractName, contractAddress);
+  }
+}
+
+export class Logger {
+  static error(message: string) {
+    console.log(`\x1b[31m[𐄂] \x1b[0m${message}`);
+  }
+
+  static success(message: string) {
+    console.log(`\x1b[32m[✓] \x1b[0m${message}`);
+  }
+
+  static warning(message: string) {
+    console.log(`\x1b[33m[⚠] \x1b[0m${message}`);
+  }
+}
+
+export class TempContractDeployments {
+  contracts: Map<string, string>;
+  filePath: string;
+
+  constructor(filePath: string) {
+    this.contracts = new Map<string, string>();
+    this.filePath = filePath;
+    this.load();
+  }
+
+  addContract(contractName: string, contractAddress: string) {
+    this.contracts.set(contractName, contractAddress);
+    this.save();
+  }
+
+  async getContract(contractName: string): Promise<Contract | null> {
+    if (!this.contracts.has(contractName)) {
+      return null;
+    }
+    const contractAddress = this.contracts.get(contractName) as string;
```

Comment:
I think it's good to additionally check if the bytecode is not empty.
You can throw or just remove` this.contracts` entry otherwise.
This is to mitigate edge cases like if you have a partial deployment and trying to proceed but occasionally switched the network.